src/Foundatio/Storage/FolderFileStorage.cs

```diff
@@ -225,9 +225,21 @@ public FolderFileStorage(Builder<FolderFileStorageOptionsBuilder, FolderFileStor
 
             var list = new List<FileSpec>();
             if (!Directory.Exists(Path.GetDirectoryName(Path.Combine(Folder, searchPattern))))
-                return Task.FromResult<IEnumerable<FileSpec>>(list);
+                return FileListResult.Empty;
 
-            foreach (string path in Directory.EnumerateFiles(Folder, searchPattern, SearchOption.AllDirectories).Skip(skip ?? 0).Take(limit ?? Int32.MaxValue)) {
+            var result = new FileListResult(() => Task.FromResult(GetFiles(searchPattern, 1, limit)));
+            await result.NextPageAsync().AnyContext();
```

Comment:
Do we need to be calling this here, we are already calling GetFiles and populating the initial result.

Reply:
Yes, it’s not being called yet. It’s passing a lamda that gets the files and it needs to call it the 1st time.

Reply:
ah, I didn't catch that, was kind of confused for a minute.